src/frontend/site.js

```diff
@@ -48,46 +48,54 @@ router.post('/bio/:username', postBio);
 //Membership
 router.get('/membership', showMembership);
 router.post('/membership/:username', postMembership);
+//Like
+router.post('/like/:postId', postLike);
 
 router.use('/ad-manager', adManagerRouter);
 
-function showGlobalTimeline(req, res) {
+async function showGlobalTimeline(req, res) {
     fetchUsingDeploymentBase(req, () =>
         Promise.all([
             req.MICROBLOG_API.get('/timeline'),
             getMembershipOfLoggedInUser(req)
         ]))
-        .then(([timeline, membership]) => {
+        .then(async ([timeline, membership]) => {
+            let postArray = timeline.data;
+            postArray = await insertLikeCountIntoPostArray(req, postArray);
+
             let data = extendRenderData({
-                data: timeline.data,
+                data: postArray,
                 title: 'Timeline',
                 username: getJwtUser(req.cookies),
                 isAdManager: hasJwtRole(req.cookies, roles.AD_MANAGER),
                 baseData: baseRequestFactory.baseData,
                 membership: membership.data.membership
 
             }, req);
-
             res.render('index.njk', data)
         }, (err) => displayError(err, res));
 }
 
-function showPersonalTimeline(req, res) {
+async function showPersonalTimeline(req, res) {
     fetchUsingDeploymentBase(req, () =>
         Promise.all([
             req.MICROBLOG_API.get('/mytimeline'),
             getMembershipOfLoggedInUser(req)
         ]))
-        .then(([myTimeline, membership]) => {
+        .then(async ([myTimeline, membership]) => {
```

Comment:
[ISSUE] Please use await instead of then and reduce the nesting level here.

Reply:
Await used and nesting level reduced